pkg/asset/machines/clusterapi.go

```diff
@@ -170,6 +172,65 @@ func GenerateClusterAPI(ctx context.Context, installConfig *installconfig.Instal
 			Object: bootstrapMachine,
 		})
 	case "azure":
+	case gcptypes.Name:
+		// Setup machine pool
+		mpool := defaultGCPMachinePoolPlatform(pool.Architecture)
+		mpool.Set(ic.Platform.GCP.DefaultMachinePlatform)
+		mpool.Set(pool.Platform.GCP)
+		if len(mpool.Zones) == 0 {
+			azs, err := gcp.ZonesForInstanceType(ic.Platform.GCP.ProjectID, ic.Platform.GCP.Region, mpool.InstanceType)
+			if err != nil {
+				return nil, fmt.Errorf("failed to fetch availability zones: %w", err)
+
+			}
+			mpool.Zones = azs
+		}
+
+		// Use the rhcosImage as the image name
+		if mpool.OSImage == nil {
+			mpool.OSImage = &gcptypes.OSImage{Name: string(*rhcosImage)}
+		}
+
+		pool.Platform.GCP = &mpool
+
+		// Separate subnets are defined for masters and workers
+		subnets := map[string]string{}
+		subnets["master"] = ic.Platform.GCP.ControlPlaneSubnet
+		subnets["worker"] = ic.Platform.GCP.ComputeSubnet
+
+		// UserLabels and UserTags are defined for the GCP Platform while the
+		// GCPMachineSpec has AdditionalLabels and AdditionalNetworkTags
+		// TODO - determine correct mapping between these
+		userTags := make(map[string]string, len(installConfig.Config.Platform.GCP.UserTags))
```

Comment:
We need to make sure that the owned tag is added as it is a default when the installer creates resources. We can find out if this is applied by CAPI or the installer should add it here.

Reply:
Done